core/src/main/kotlin/in/specmatic/conversions/BearerSecurityScheme.kt

```diff
@@ -9,7 +9,9 @@ import org.apache.http.HttpHeaders.AUTHORIZATION
 
 class BearerSecurityScheme : OpenAPISecurityScheme {
     override fun matches(httpRequest: HttpRequest): Boolean {
-        return httpRequest.headers.containsKey(AUTHORIZATION)
+        return httpRequest.headers.let {
+            it.containsKey(AUTHORIZATION) && it[AUTHORIZATION]!!.lowercase().startsWith("bearer")
```

Comment:
@joelrosario Would that be something like this? It is rather awkward to not get a better error than the generic `No auth params were found in the request`, but I'm not quite sure the best way to change the signature of `matches()` in a way it can create different failure responses.

Reply:
@androa yes, that change looks good, but you're also right about the generic message not being too helpful.

We'd use Specmatic's Result class to provide more detail.

This means, we'd need to change the return type of the `matches` method from Boolean to Result.

I believe there should not be much impact as this method is not referred to in many places, which include:
1. the other classes implementing OpenAPISecurityScheme
2. `HttpRequestPattern.matchSecurityScheme`... we'd need to modify a line of code in that method to match the new signature.

Reply:
Made an attempt of refactoring it into returning the Result class now.

Reply:
I'm having some issues with running the test suite locally so the failing test in OpenApiKtTest slipped past me. Not quite sure what causes it though.